### Verification — `9e6d767` (rev 1)

| # | Claim | Result |
| --- | --- | --- |
| 1 | 새 규칙이 지목한 vacuity 가 실재한다 — 양쪽이 명령치환인 비교는 대상이 사라져도 exit 0 이다 | PASS(live) |
| 2 | 처방이 실제로 듣는다 — 한쪽을 리터럴로 고정하면 같은 조건에서 exit 1 이 된다 | PASS(live) |
| 3 | 이 형태는 가설이 아니다 — 실제 spec store 의 Verify 18줄 중 1줄이 그 형태이고, 대상을 치우면 양쪽 다 에러를 내면서 exit 0 이다 | PASS(live) |
| 4 | 이 PR 의 고유 변경은 문서 2개뿐이다 — diff 의 나머지는 스택 때문에 보이는 #1018 분량이다 | PASS(live) |

<details><summary>Unverified — (a) 새 규칙이 첫 규칙과 충돌하지 않는다는 문서의 주장은 설계 논증이라 실행으로 갈리지 않음, (b) store 에서 발견된 그 1줄은 이 PR 이 고치지 않음</summary>
<br>

**(a)** 문서는 새 규칙과 기존 첫 규칙("종료 코드는 환경이 아니라 요구사항을 보고해야 한다")이 충돌하지 않는다고 적고, 근거로 범위 차이를 듭니다 — 첫 규칙은 체크아웃 *밖* per-user 상태, 새 규칙은 요구사항이 실제로 단언하는 repo 상대 입력. 두 규칙이 각각 무엇을 잡는지는 행 1~3 에서 실행으로 확인했지만, "그러므로 충돌하지 않는다" 는 경계 판단이라 참·거짓이 실행으로 갈리지 않습니다.

**(b)** 행 3 이 찾아낸 줄은 `<STORE>/0001-spec-store.md:71` 입니다. 이 PR 은 규칙을 문서화할 뿐 그 줄을 고치지 않으며, 대상이 저장소 밖 per-user store 라 애초에 이 PR 의 diff 로 고칠 수 있는 것도 아닙니다. 후속 판단이 필요한 항목으로 남깁니다 — 지금 여기서 이슈를 만들지 않은 것은 접근안 리뷰가 먼저이기 때문입니다.

</details>

<details><summary>Evidence 1 — 대상이 있는 경우와 없는 경우를 같은 두 형태로 대조</summary>
<br>

규칙이 하는 말은 "이 형태는 실패할 수 없다" 입니다. 부재 주장이므로 양성 대조가 먼저 필요합니다 — 대상이 있을 때 두 형태가 **모두** 통과해야, 부재 시의 차이가 형태 때문이라고 말할 수 있습니다.

```
$ bash "$P/probe_1019.sh"
== 1. subject PRESENT (positive control — both forms must agree)
    both-substitutions : exit 0  (implemented)
    one-side-literal   : exit 0  (implemented)

== 2. subject DELETED (the case the rule is about)
    both-substitutions : exit 0  (implemented)  <- VACUOUS
    one-side-literal   : exit 1  (missing)
```

대상이 있을 때 두 형태가 같은 답을 내고, 대상을 지우면 갈립니다. 양쪽이 명령치환인 쪽은 둘 다 빈 문자열로 무너져 `test "" = ""` 가 되고 exit 0 입니다 — 아무것도 만족시키지 않는 요구사항이 `implemented` 로 보고됩니다.

`$P` 는 프로브를 워크트리 밖에 두려고 담아 둔 경로 변수입니다.

</details>

<details><summary>Evidence 2 — 한쪽을 리터럴로 고정한 형태가 같은 조건에서 부재를 잡음</summary>
<br>

행 2 는 위 블록의 아래쪽 절반을 읽은 것입니다 — 별도 실행이 아닙니다. 규칙이 제시하는 처방("한쪽을 리터럴로 고정하라")이 말만이 아니라 실제로 판정을 바꾸는지가 질문이고, 대상 삭제 후 `one-side-literal` 만 `exit 1` 로 넘어갑니다.

```
== 2. subject DELETED (the case the rule is about)
    both-substitutions : exit 0  (implemented)  <- VACUOUS
    one-side-literal   : exit 1  (missing)      <- caught
```

리터럴 쪽은 대상이 없어도 비교 상대가 남아 있어 불일치가 드러납니다. 이것이 처방의 전부이고, 규칙이 요구하는 것도 그 이상이 아닙니다.

</details>

<details><summary>Evidence 3 — 실제 spec store 에서 그 형태를 찾아, 그 줄 자체를 대상 없이 실행</summary>
<br>

규칙이 잡으려는 형태가 실제로 존재하지 않는다면 이 문서 변경은 예방이지 교정이 아닙니다. 그래서 실제 store 의 Verify 줄을 전수로 훑고, 걸린 줄을 **그 줄 그대로** 두 조건에서 돌렸습니다.

```
$ bash "$P/probe_1019.sh"
== 3. does the shape occur in the repo's own spec store?
    store: /Users/nathan.song/.praxis/docs/specs
    Verify lines: 18   both-sides-substitution: 1
```

걸린 줄은 `<STORE>/0001-spec-store.md:71` — `praxis_specs_dir()` 의 셸 구현과 파이썬 구현을 비교하는 요구사항이고, 양쪽 모두 명령치환입니다. 체크아웃 안에서(대상 있음)와 빈 디렉터리에서(대상 없음) 각각 돌립니다:

```
$ bash "$P/probe_1019b.sh"
== subject PRESENT (run from the checkout — positive control)
    exit=0  -> implemented

== subject GONE (run from an empty dir — hooks/_lib/_paths.sh does not exist)
    exit=0
    stderr: sh: hooks/_lib/_paths.sh: No such file or directory
    stderr: Traceback (most recent call last):
    stderr:   File "<string>", line 1, in <module>
```

**양쪽이 모두 실패했는데 exit 0 입니다.** 셸은 파일을 못 찾고 파이썬은 traceback 을 냈는데도 두 치환이 빈 문자열로 무너져 비교가 성립했습니다 — 보고서는 이것을 `implemented` 로 찍습니다. 규칙이 서술한 그대로가 store 안에 살아 있습니다. store 경로의 홈 부분은 위 출력에서 프로브가 그대로 찍은 것이고, 인용할 때만 `<STORE>` 로 줄였습니다.

</details>

<details><summary>Evidence 4 — 스택 때문에 부풀어 보이는 diff 와 실제 커밋 범위를 분리</summary>
<br>

이 PR 의 base 는 원래 #1018 의 브랜치였고, 부모를 `--delete-branch` 로 머지하면 자식이 auto-close 되므로 base 를 `main` 으로 재타깃했습니다. 그 결과 diff 에 #1018 의 변경(`spec_drift.py` 의 `stdin=DEVNULL`, 케이스 20, `runtime-verified-note` 갱신)이 함께 보입니다 — 이 PR 이 그것들을 주장하는 것이 아닙니다.

```
$ git log --oneline origin/main..HEAD
9e6d767 docs(spec): a Verify oracle must fail when its inputs are gone
6e6c4ee fix(spec-drift): run Verify commands with stdin at /dev/null

$ git show --stat --oneline HEAD
9e6d767 docs(spec): a Verify oracle must fail when its inputs are gone
 docs/spec-store.md         | 14 +++++++++++++-
 skills/spec-drift/SKILL.md |  8 +++++++-
 2 files changed, 20 insertions(+), 2 deletions(-)
```

고유 커밋은 하나, 문서 2개, 순증 20줄입니다. `6e6c4ee` 는 #1018 의 것이고 그쪽 앵커에서 검증했습니다. **머지 순서는 #1018 이 먼저입니다** — 이 PR 을 먼저 머지하면 #1018 의 변경이 함께 들어가고 #1018 은 빈 PR 이 됩니다.

</details>

<details><summary>History — rev 1, 최초 게시</summary>
<br>

이전 판본 없음. 대체된 근거도 없습니다.

</details>

--- a/docs/spec-store.md
+++ b/docs/spec-store.md
@@ -162,7 +162,8 @@ not run. Writing a second `Verify:` in one block is an authoring error — the
 first binds, and the report prints the rest as `warning` lines naming the
 command it did not run.
 
-Two rules on what may appear there:
+Rules on what may appear there — the count is deliberately not stated, so that
+adding one cannot leave a stale number behind:
 
 - **The exit code must report the requirement, not the environment.** A command
   that can exit non-zero for a reason outside the repository is not eligible —
@@ -174,6 +175,17 @@ Two rules on what may appear there:
   so a command that reads stdin gets immediate EOF. Before that it inherited the
   report's own stdin, hung until the timeout, and was reported `missing` — a
   rule stated here but not enforced showed up only as a wrong verdict.
+- **The command must fail when its inputs are gone.** An oracle whose subject
+  has been deleted must exit non-zero, not collapse into a vacuous success. The
+  shape that breaks this is a comparison whose *both* sides are command
+  substitutions: when the subject is absent both substitutions fail, both
+  collapse to the empty string, and `test "" = ""` exits 0 — so the report
+  prints `implemented` for a requirement nothing in the tree satisfies
+  ([#1011](https://github.com/devseunggwan/praxis/issues/1011)). Pin at least
+  one side to a literal, so an absent subject compares against something and
+  the mismatch is visible. This does not conflict with the first rule: that one
+  is about per-user state *outside* the checkout, this one is about the
+  repo-relative inputs the requirement is actually asserting.
 
 Commands run from the repository root of wherever the report was invoked, so a
 `Verify:` line may use repo-relative paths — and a spec written for repository

--- a/skills/spec-drift/SKILL.md
+++ b/skills/spec-drift/SKILL.md
@@ -110,13 +110,19 @@ checkable requirements, which is the feedback loop this design is built on.
 
 ## Writing a Verify line
 
-The convention and its two rules live in
+The convention and its rules live in
 [`docs/spec-store.md`](../../docs/spec-store.md) → *Verification lines*. In
 short: a nested `- Verify: \`<command>\`` item under the requirement, whose
 exit code reports **the requirement** and not the environment it ran in, and
 which terminates without input. Commands run with stdin at `/dev/null` (#1008),
 so one that reads stdin gets EOF rather than hanging to the timeout and being
 reported `missing`.
+
+The oracle must also **fail when its subject is gone** (#1011). A comparison
+whose *both* sides are command substitutions cannot do that: with the subject
+absent both collapse to the empty string and `test "" = ""` exits 0, so the
+report prints `implemented` for a requirement nothing satisfies. Pin one side
+to a literal.
 
 A requirement with no eligible command keeps no `Verify:` line and says why in
 its own prose — that line is what separates "nobody got to it" from "nothing

